File: android/app/src/main/java/com/lightningkite/kwifttemplate/layouts/FloatingHexagonsDemoXml.kt
//
// FloatingHexagonsDemoXml.swift
// Created by Kwift XML Android
//
package com.lightningkite.kwifttemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.kwift.views.*
import com.lightningkite.kwifttemplate.R

class FloatingHexagonsDemoXml {

    lateinit var customView: com.lightningkite.kwift.views.CustomView
    lateinit var customViewDelegate: com.lightningkite.kwifttemplate.vg.FloatingHexagonsDelegate
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.floating_hexagons_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        customView = view.findViewById<com.lightningkite.kwift.views.CustomView>(R.id.customView)
        customViewDelegate = view.findViewById<CustomView>(R.id.customView).delegate as com.lightningkite.kwifttemplate.vg.FloatingHexagonsDelegate
        
        return view
    }
}

File: ios/Kwift Template/swiftResources/layouts/FloatingHexagonsDemoXml.swift
//
// FloatingHexagonsDemoXml.swift
// Created by Kwift XML
//

import UIKit
import Kwift
import MapKit

public class FloatingHexagonsDemoXml {
    
    public unowned var xmlRoot: UIView!
    public func setup(_ dependency: ViewDependency) -> UIView {
        let view = FrameLayout(frame: .zero)
        view.padding = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
        view.addSubview(
            CustomView(frame: .zero),
            minimumSize: CGSize(width: 0, height: 0),
            size: CGSize(width: 0, height: 0),
            margin: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0),
            padding: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0),
            gravity: .fillFill
        ) { view in 
            self.customView = view
            view.backgroundColor = ResourcesColors.black
            let dg = FloatingHexagonsDelegate()
            view.delegate = dg
            self.customViewDelegate = dg
        }
        
        xmlRoot = view
        return view
    }
    
    public unowned var customViewDelegate: FloatingHexagonsDelegate!
    public unowned var customView: CustomView!
    
}
